src/blockchain/blockchain_behavior.cpp

```diff
@@ -58,6 +71,19 @@ const Settings &Behavior::GetDefaultSettings() const {
   return m_parameters.default_settings;
 }
 
+std::size_t Behavior::GetTransactionWeight(const CTransaction &tx) const {
+  return ::GetSerializeSize(tx, SER_NETWORK, PROTOCOL_VERSION | SERIALIZE_TRANSACTION_NO_WITNESS) * (m_parameters.witness_scale_factor - 1) + ::GetSerializeSize(tx, SER_NETWORK, PROTOCOL_VERSION);
```

Comment:
Can you please explain me why `witness_scale_factor - 1`?

Reply:
Version 2 in the following posting in the bitcoin mailing list: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2017-January/013504.html

Also this code is almost a verbatim copy from the existing validation.

I added a comment referring to that posting on the `witness_scale_factor` parameter which was indeed missing a description before in https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2017-January/013504.html

I will add a comment to this function also.